smartsim/_core/launcher/step/step.py

```diff
@@ -66,10 +67,20 @@ def _create_unique_name(entity_name: str) -> str:
         step_name = entity_name + "-" + get_base_36_repr(time.time_ns())
         return step_name
 
+    @staticmethod
+    def ensure_output_directory_exists(output_dir: str) -> None:
```

Comment:
- Add short docstring
- Pedantic: This function should live with a `utils`-like module. If we don't have one, then don't worry about putting in there, otherwise it would be worth doing for cleanliness

Reply:
Docstring added! There is a `util` module but nothing in there related to `step`, so I'm leaving this function in `Step.py`.